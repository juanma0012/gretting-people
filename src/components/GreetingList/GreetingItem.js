import React from 'react'
import PropTypes from 'prop-types'

const GreetingItem = ({ text }) => (
    <li className="greetingItem">{text}</li>
)

GreetingItem.propTypes = {
    text: PropTypes.string.isRequired
}

export default GreetingItem
