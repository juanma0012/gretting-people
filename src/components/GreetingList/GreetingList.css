.GreetingList{
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
}
.GreetingList > ul {
    text-align: left;
}

.GreetingList > ul li {
  font-size: 1em;
  color: #535353;
  padding: 5px;
}