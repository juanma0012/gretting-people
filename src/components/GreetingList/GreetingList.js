import { connect } from 'react-redux'
import React from 'react'
import PropTypes from 'prop-types'

const GreetingList = ({ greetings }) => {
    const greetingAllPeople = greetings.map((greeting, i) => (
        <li key={i}>{greeting}</li>
    ))

    return (
        <ul>
            {greetingAllPeople}
        </ul>
    )
}
GreetingList.propTypes = {
    greetings: PropTypes.array.isRequired
}

const mapStateToProps = (state) => ({
    greetings: state.greetingList.greetings
})

export default connect(
    mapStateToProps
)(GreetingList)
