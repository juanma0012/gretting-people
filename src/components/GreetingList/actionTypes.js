export const SET_GREETINGS = 'greetingList/SET_GREETINGS';
