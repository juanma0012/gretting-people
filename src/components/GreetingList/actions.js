import { greetingByLanguage } from '../../shared/utility';
import * as actionTypes from './actionTypes';

export const setGreetings = (people, language) => {
    let greetings = people.map(person => greetingByLanguage[language](person));
    return { type: actionTypes.SET_GREETINGS, greetings };
};
