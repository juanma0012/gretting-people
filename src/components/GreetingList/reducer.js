import { resolveEach } from 'redux-resolver'
import { updateObject } from '../../shared/utility';
import * as actionTypes from './actionTypes'

const initialState = {
    greetings: []
};

const setGreetings = (state, action) => {
    console.log("action", action);
    return updateObject(state, {
        greetings: action.greetings
    });
};

export default resolveEach(initialState, {
    [actionTypes.SET_GREETINGS]: setGreetings
});
