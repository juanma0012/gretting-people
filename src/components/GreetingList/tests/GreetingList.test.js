import React from 'react';
import { shallow } from 'enzyme';
import { GreetingList } from '../GreetingList';
import GreetingItem from '../GreetingItem';

describe('Component GreetingList ', () => {
    const list = [
        'First greeting',
        'Second greeting'
    ]
    const component = shallow(<GreetingList greetings={list} />);

    describe('WHEN is initialized ', () => {
        it('SHOULD render correctly ', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('WHEN receive a list of greetings ', () => {
        it('SHOULD render all the greetings ', () => {
            expect(component.find(GreetingItem)).toHaveLength(2);
        });
    });
});
