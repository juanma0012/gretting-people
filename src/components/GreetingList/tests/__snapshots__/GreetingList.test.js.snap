// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GreetingList  WHEN is initialized  SHOULD render correctly  1`] = `
<div
  className="greetingList"
>
  <ul>
    <GreetingItem
      key="0"
      text="First greeting"
    />
    <GreetingItem
      key="1"
      text="Second greeting"
    />
  </ul>
</div>
`;
